feat: issue_04

yoshikawa-akr opened the pull request:
# [issue No.4](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#4)

**issue**
```
[{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}]
```
という配列を使って以下のようなHTML出力にしてください
```
<ul>
 <li><a href="/bookmark.html"><img src="1.png" alt="画像1">ブックマーク</a></li>
 <li><a href="/message.html"><img src="2.png" alt="画像2">メッセージ</a></li>
</ul>
```

## [CodeSandBox](https://codesandbox.io/s/serene-montalcini-iohlmg) / Latest commit: https://github.com/yoshikawa-akr/js-lessons/pull/4/commits/1a9f5eee11d08f2372bca93c6000c84d3cf9eb96

## In terms of implementation, what you are worried about, what you want to focus on, and what you want to ask.
09e6c6c3474b8d48070835e761e7be13657cc2cf
I tried using "for of" for loop processing.

yoshikawa-akr committed:
feat: Create the specified array.

yoshikawa-akr committed:
feat: Add processing for issue 4.

seeeen044 commented on issue_04/script.js:
```diff
@@ -0,0 +1,28 @@
+
+    fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin', img);
+}
+document.body.insertAdjacentElement('afterbegin', ul).appendChild(fragment);
```
Question

Why did you omit HTML ul tag and add DOM to the body tag?

yoshikawa-akr replied:
I thought the task was to build the DOM in JavaScript, including the ul tag.

seeeen044 replied:
Thank you for the answer.
As a code, I thought it would be easier to read if you installed ul like in issue 3.

hippopo-moo commented on issue_04/script.js:
```diff
@@ -0,0 +1,28 @@
+const contentsArray = [{
```
IMO
You can tell from the value that it is an array, so the variable name might as well be "contents".

yoshikawa-akr replied:
It's true that the value shows that it is an array, so we should review the array name. 
Thank you very much.
https://github.com/yoshikawa-akr/js-lessons/pull/4/commits/5febcd6735a7b1bad6987907b8ec43f34b721951

yoshikawa-akr committed:
refactor: Renaming Arrays.

yoshikawa-akr committed:
refactor: Add ul tag to the HTML file.

Changed the process to write the ul tag directly in index.html and retrieve it with JavaScript.
